src/ObjCRuntime/Registrar.cs

```diff
@@ -2163,6 +2165,10 @@ protected string ToSignature (TType type, ObjCMember member, ref bool success, b
 				throw CreateException (4102, member, "The registrar found an invalid type `{0}` in signature for method `{2}`. Use `{1}` instead.", "System.DateTime", IsDualBuild ? "Foundation.NSDate" : CompatNamespace + ".Foundation.NSDate", member.FullName);
 			}
 
+			// We use BindAsAttribute to wrap NSNumber/NSValue into more accurate Nullable<T> types
+			if (typeFullName != null && typeFullName.Contains ("Nullable"))
```

Comment:
Contains is not safe enough 
Startwith would be better
But it might be possible to check for generic earlier